crowbar_framework/lib/repositories.rb

```diff
@@ -0,0 +1,126 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Repositories
+  class Check
+
+    def initialize(options = {})
+      @platform = options.fetch :platform
+      @name = options.fetch :name
+    end
+
+    def validate
+      ret = {}
+      ret[:dir_available] = validate_repo_dir
+      ret[:key_md5] = validate_key_file
+      ret[:required] = requirement
+      ret[:web_url] = web_url
+      ret
+    end
+
+    protected
+
+    #
+    # config
+    #
+    def load_config
+      config_path = File.join(Rails.root, "config", "repos.yml")
+      YAML.load_file(config_path)
+    end
+
+    def repo_config
+      load_config[@platform]["repos"][@name.downcase]
+    end
+
+    def md5_repomd
+      repo_config["repomd"]["md5"]
+    end
+
+    def repomd_key
+      repo_config["repomd"]["key"] || "/srv/tftpboot/#{@platform}/repos/#{@name}/repodata/repomd.xml.key"
+    end
+
+    def requirement
+      repo_config["required"]
+    end
+
+    def web_url
+      repo_config["url"] || \
+      "http://#{Repositories.admin_ip}:#{Repositories.web_port}/#{@platform}/repos/#{@name}"
```

Comment:
Style/MultilineOperationIndentation: Use 2 (not 0) spaces for indenting an expression spanning multiple lines.